drivers/at/at.c

```diff
@@ -128,17 +131,21 @@ ssize_t at_send_cmd_get_lines(at_dev_t *dev, const char *command,
     }
 
     while (1) {
-        res = at_readline(dev, pos, bytes_left, timeout);
+        res = at_readline(dev, pos, bytes_left, keep_eol, timeout);
         if (res == 0) {
+            if (bytes_left) {
+                *pos++ = eol[sizeof(eol) - 2];
+                bytes_left--;
+            }
             continue;
         }
         else if (res > 0) {
             bytes_left -= res;
-            if ((res == 2) && (strncmp(pos, "OK", 2) == 0)) {
+            if ((res == (2 + (keep_eol ? 1 : 0))) && (strncmp(pos, "OK", 2) == 0)) {
```

Comment:
here and below: if ```keep_eol```, it has values 0 or 1, thus the ternary operator can be removed. e.g., just use ```2 + keep_eol)```

Reply:
changed